### PR TITLE
Deflake managedresource controller integration test

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area testing dev-productivity
/kind flake cleanup

**What this PR does / why we need it**:

Fixes several flakes in the managedresource controller integration test, that were caused by changes in https://github.com/gardener/gardener/pull/5904.
In essence, the test cases were not able to cater with faster reconcile cycles, which resulted in conflicts, race conditions, etc. After lowering the sync period in https://github.com/gardener/gardener/pull/5904 to speed up the integration test these problems surfaced in CI. Previously we were just lucky / they were unlikely to surface because of the high sync period and long wait timeouts.

In order to reproduce, debug and fix the flakes, I had to rework the test suite significantly, so that it can be executed in parallel against an existing cluster.
I simplified and enhanced the test code significantly along the way, as well as fixed a wrongful test case for the `DeleteOnInvalidUpdate` annotation.

**Which issue(s) this PR fixes**:

Fixes 4 kinds of flakes:

- flake 1: `should not reconcile ManagedResource of any other class except the default class`, observed in https://prow.gardener.cloud/view/gs/gardener-prow/logs/ci-gardener-integration/1522509775537967104
  - fixed by the refactoring to use dedicated resource names per test case
- flake 2: `should set ManagedResource to unhealthy`, observed in https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/5935/pull-gardener-integration/1522480867149090816 and https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/5938/pull-gardener-integration/1523470537722957824
- flake 3: different test failures caused by conflicts errors, observed in https://prow.gardener.cloud/view/gs/gardener-prow/logs/ci-gardener-integration/1522992965088186368 and https://prow.gardener.cloud/view/gs/gardener-prow/pr-logs/pull/gardener_gardener/5926/pull-gardener-integration/1523542805262110720
- flake 4: deletion times out because of constant requeue on pending deletion, observed locally

With this PR, the test suite looks much more stable:
```
$ make kind-up
$ export USE_EXISTING_CLUSTER=true
$ export GOMEGA_DEFAULT_EVENTUALLY_TIMEOUT=5s GOMEGA_DEFAULT_EVENTUALLY_POLLING_INTERVAL=200ms GOMEGA_DEFAULT_CONSISTENTLY_DURATION=5s GOMEGA_DEFAULT_CONSISTENTLY_POLLING_INTERVAL=200ms
$ cd ./test/integration/resourcemanager/managedresource && ginkgo build . && stress -p 32 ./managedresource.test
...
16m40s: 452 runs so far, 0 failures
```

**Special notes for your reviewer**:

The first 6 commits focus on reworking the test suite to make flakes reproducible with `stress` and `USE_EXISTING_CLUSTER=true`.
The last 3 commits fix individual flakes mentioned above.

### CODE REVIEW

**Comment by timebertt** [comment]
/cc @acumino @kris94 
as original authors of the integration test.

File: test/integration/resourcemanager/managedresource/resource_test.go [hunk 1]
```diff
@@ -786,10 +786,10 @@ var _ = Describe("ManagedResource controller tests", func() {
 				updatedDeployment.Spec.Template = *newPodTemplateSpec
 				Expect(testClient.Update(ctx, updatedDeployment)).To(Succeed())
 
-				Consistently(func(g Gomega) bool {
+				Eventually(func(g Gomega) corev1.ResourceRequirements {
```
Comment (shafeeqes) [inline test/integration/resourcemanager/managedresource/resource_test.go]: Shouldn't we use `Consistently` itself here?

**Review by shafeeqes — COMMENTED** [review]
Thanks for the well structured PR and the commit messages.

File: test/integration/resourcemanager/managedresource/resource_test.go (hunk 1 above)
Comment (timebertt) [reply test/integration/resourcemanager/managedresource/resource_test.go]: Good catch, thank you. Changed it back.

**Pull request merged**